packages/runtime/datastore/src/localChannelStorageService.ts

```diff
@@ -58,3 +58,56 @@ export class LocalChannelStorageService implements IChannelStorageService {
         return undefined;
     }
 }
+
+// Some interface (as an example) that implements readBlob.
+// This can be IDocumentStorage, IChannelStorageService, etc.
+export interface IFoo {
+    someOtherMethods(): void;
+
+    readBlob(blobid: string): IsoBuffer;
+
+    // This is optional (i.e. may not be on interface), but it's nice to have it here.
+    // It basically says that underlying interface (like IDocumentStorage) could imlement readString()
+    // if it wants to. It does not have to, but it may chose to do so if it has extra knowlege allowing
+    // it to implement such functionality more efficiently
+    // I.e. SPO driver may implement it because it already operates (for most part) with strings, going through
+    // readBlob() would add more conversions, and thus make it slower.
+    readString?(blobId: string): string;
+}
+
+// For each such interface, we can define an extended interface that has readString()
+// We will demostrate below how to have one shared implementaiton that can add readString()
+// to any interface that implements readBlob().
+export interface IFooEx extends IFoo {
+    readString(blobId: string): string;
+}
+
+// This is input to addReadString() - it basically defined any type that has readBlob
+// Any object that satisfy this interace (like IFoo) will be accespted
+export interface IHasReadBlob {
+    readBlob(blobid: string): IsoBuffer;
+}
+
+// This is what we add to such object.
+export interface IHasReadString {
+    readString(blobid: string): string;
+}
+
+// Function that takes any object that implements readBlob() and adds implementation of readString.
+export function addReadString<T extends IHasReadBlob>(object: T): T & IHasReadString {
+    const objectEx = object as T & IHasReadString;
+
+    // check if readString is already implemented. If it is, nothing to do.
+    // We are better off leaving it as is, as it can be implemented in more efficient way.
+    if (objectEx.readString !== undefined) {
+        return objectEx;
+    }
+
+    // make a copy and add method
+    const copy = Object.create(objectEx) as T & IHasReadString;
```

Comment:
Note that we are created a copy of the object. It's shallow copy, but nonetheless it's a copy.
The benefit of doing so is that it's cleaner - we are not modifying original object.
The con - if we keep calling this function for original object 100 times, we will get 100 copies. If we were not copying here but modifying object in-place, then calling it 100 times would not create any copy.
I prefer copy, but it also means we should strive to call this function once for each object.